tests/Aspire.Hosting.Nats.Tests/NatsFunctionalTests.cs

```diff
@@ -0,0 +1,244 @@
+                    await app.StopAsync();
+                }
+            }
+
+            var builder2 = CreateDistributedApplicationBuilder();
+            var nats2 = builder2.AddNats("nats")
+                .WithJetStream();
+
+            if (useVolume)
+            {
+                nats2.WithDataVolume(volumeName);
+            }
+            else
+            {
+                nats2.WithDataBindMount(bindMountPath!);
+            }
+
+            using (var app = builder2.Build())
+            {
+                await app.StartAsync();
+                try
+                {
+                    var hb = Host.CreateApplicationBuilder();
+
+                    hb.Configuration.AddInMemoryCollection(new Dictionary<string, string?>
+                    {
+                        [$"ConnectionStrings:{nats2.Resource.Name}"] = await nats2.Resource.ConnectionStringExpression.GetValueAsync(default)
+                    });
+
+                    hb.AddNatsClient("nats", configureOptions: opts =>
+                    {
+                        var jsonRegistry = new NatsJsonContextSerializerRegistry(AppJsonContext.Default);
+                        return opts with { SerializerRegistry = jsonRegistry };
+                    });
+
+                    hb.AddNatsJetStream();
+
+                    using (var host = hb.Build())
+                    {
+                        await host.StartAsync();
+
+                        var pipeline = new ResiliencePipelineBuilder()
+                            .AddRetry(new() { MaxRetryAttempts = 10, Delay = TimeSpan.FromSeconds(1), ShouldHandle = new PredicateBuilder().Handle<NatsException>() })
+                            .Build();
+                        await pipeline.ExecuteAsync(async token =>
+                        {
+                            var jetStream = host.Services.GetRequiredService<INatsJSContext>();
+
+                            var stream = await jetStream.GetStreamAsync(streamName, cancellationToken: token);
+                            var consumer = await stream.CreateOrderedConsumerAsync(cancellationToken: token);
+
+                            var events = new List<AppEvent>();
+                            await foreach (var msg in consumer.ConsumeAsync<AppEvent>(cancellationToken: token))
+                            {
+                                events.Add(msg.Data!);
+
+                                if (msg.Metadata?.NumPending == 0)
+                                {
+                                    break;
+                                }
+                            }
+
+                            foreach (var item in events.Select((appEvent, i) => new { i, appEvent }))
+                            {
+                                Assert.Equal($"test-event-{item.i}", item.appEvent.Name);
+                                Assert.Equal($"test-event-description-{item.i}", item.appEvent.Description);
+                            }
```

Comment:
This should verify the expected number came through. We should convert it to `for (var i = 0; i < 10; i++)`. That way if less than 10 come through, the test fails.

Reply:
That would help. But using a for loop would elimitate the Select and anonymous type that is happening here.